internal/raft/message/convert.go

```diff
@@ -0,0 +1,971 @@
+	msgListOffset.Offset.Offset = ConvertCommandToMessageExpr(cmd.Offset)
+	msgListOffset.Offset.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListOffset
+}
+
+// ConvertCommandToMessageListDistinct converts a command.Distinct to a message.List_Distinct
+func ConvertCommandToMessageListDistinct(cmd *command.Distinct) *List_Distinct {
+	msgListDistinct := &List_Distinct{&Command_Distinct{}}
+	msgListDistinct.Distinct.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListDistinct
+}
+
+// ConvertCommandToMessageRepeatedExprSlice converts a [][]command.Expr to a [][]message.Expr
+func ConvertCommandToMessageRepeatedExprSlice(cmd [][]command.Expr) []*RepeatedExpr {
+	msgRepeatedExprSlice := []*RepeatedExpr{}
+	for i := range cmd {
+		msgRepeatedExpr := &RepeatedExpr{}
+		for j := range cmd[i] {
+			msgRepeatedExpr.Expr = append(msgRepeatedExpr.Expr, ConvertCommandToMessageExpr(cmd[i][j]))
+		}
+		msgRepeatedExprSlice = append(msgRepeatedExprSlice, msgRepeatedExpr)
+	}
+	return msgRepeatedExprSlice
+}
+
+// ConvertCommandToMessageListValues converts a command.Values to a message.List_Values
+func ConvertCommandToMessageListValues(cmd *command.Values) *List_Values {
+	msgListValues := &List_Values{}
+	msgListValues.Values.Expr = ConvertCommandToMessageRepeatedExprSlice(cmd.Values)
+	return msgListValues
+}
+
+// ConvertCommandToMessageList converts
+func ConvertCommandToMessageList(cmd command.List) *List {
+	msgList := &List{}
+	switch cmd := cmd.(type) {
+	case *command.Scan:
+		msgList.List = ConvertCommandToMessageListScan(cmd)
+	case *command.Select:
+		msgList.List = ConvertCommandToMessageListSelect(cmd)
+	case *command.Project:
+		msgList.List = ConvertCommandToMessageListProject(cmd)
+	case *command.Join:
+		msgList.List = ConvertCommandToMessageListJoin(cmd)
+	case *command.Limit:
+		msgList.List = ConvertCommandToMessageListLimit(cmd)
+	case *command.Offset:
+		msgList.List = ConvertCommandToMessageListOffset(cmd)
+	case *command.Distinct:
+		msgList.List = ConvertCommandToMessageListDistinct(cmd)
+	case *command.Values:
+		msgList.List = ConvertCommandToMessageListValues(cmd)
+	}
+	return msgList
+}
+
+// ConvertCommandToMessageSelect converts a Command type to a Command_Select type.
+func ConvertCommandToMessageSelect(cmd command.Command) *Command_Select {
+	msgCmdSelect := &Command_Select{}
+	msgCmdSelect.Filter = ConvertCommandToMessageExpr(cmd.(*command.Select).Filter)
+	msgCmdSelect.Input = ConvertCommandToMessageList(cmd.(*command.Select).Input)
+	return msgCmdSelect
+}
+
+// ConvertCommandToMessageCol converts command.Column to a message.Column
+func ConvertCommandToMessageCol(cmd command.Column) *Column {
+	msgCol := &Column{}
+	msgCol.Table = cmd.Table
+	msgCol.Column = ConvertCommandToMessageExpr(cmd.Column)
+	msgCol.Alias = cmd.Alias
+	return msgCol
+}
+
+// ConvertCommandToMessageCols converts []command.Column to a []message.Column
+func ConvertCommandToMessageCols(cmd []command.Column) []*Column {
```

Comment:
Probably `ConvertCommands` (plural) is more appropriate here?

Reply:
Ill make it `ColSlice`